### PR TITLE
(PA-5733) Enable puppet-runtime builds for rhel9

### PR DESCRIPTION
**Pull request by amitkarsale**
Added platform definition of redhat9 to enable puppet-runtime build.

### CODE REVIEW

**Review by joshcooper — APPROVED** [review]
@amitkarsale looks good for next time could you include a link to the vanagon generic build for agent-runtime-7.x and agent-runtime-main? Also would be good to mention here or in the ticket that installing `swig` was blocked on an RE-15758

**Review by joshcooper — CHANGES_REQUESTED** [review]
Oh wait, there are build failures. I think you need to require `perl` not just `perl-Getopt-Long`:

```
 ./Configure --prefix=/opt/puppetlabs/puppet --libdir=lib --openssldir=/opt/puppetlabs/puppet/ssl shared no-asm linux-aarch64 no-camellia no-ec2m no-md2 no-ssl3 no-dtls no-dtls1 no-idea no-seed no-weak-ssl-ciphers -DOPENSSL_NO_HEARTBEATS -I/opt/puppetlabs/puppet/include -I/opt/pl-build-tools/include -L/opt/puppetlabs/puppet/lib -L/opt/pl-build-tools/lib -Wl,-rpath=/opt/puppetlabs/puppet/lib -Wl,-z,relro
Can't locate FindBin.pm in @INC (you may need to install the FindBin module) (@INC contains: /usr/local/lib64/perl5/5.32 /usr/local/share/perl5/5.32 /usr/lib64/perl5/vendor_perl /usr/share/perl5/vendor_perl /usr/lib64/perl5 /usr/share/perl5) at ./Configure line 15.
BEGIN failed--compilation aborted at ./Configure line 15.
make: *** [Makefile:97: openssl-1.1.1-configure] Error 2
```

**Comment by amitkarsale** [comment]
> @amitkarsale looks good for next time could you include a link to the vanagon generic build for agent-runtime-7.x and agent-runtime-main? Also would be good to mention here or in the ticket that installing `swig` was blocked on an RE-15758

Yes @joshcooper , I didnt mentioned that there is a blocker on swig package.  I'll keep this in mind. 

Here is the build link for main : https://jenkins-platform.delivery.puppetlabs.net/job/platform_vanagon-generic-builder_vanagon-packaging_generic-builder/2328/
 
Here is the build link for 7.x : https://jenkins-platform.delivery.puppetlabs.net/job/platform_vanagon-generic-builder_vanagon-packaging_generic-builder/2329/

Do we still need `perl` package?

**Comment by amitkarsale** [comment]
> @amitkarsale looks good for next time could you include a link to the vanagon generic build for agent-runtime-7.x and agent-runtime-main? Also would be good to mention here or in the ticket that installing `swig` was blocked on an RE-15758

Yes @joshcooper , I didnt mentioned that there is a blocker on swig package. I'll keep this in mind.

Here is the build link for main : https://jenkins-platform.delivery.puppetlabs.net/job/platform_vanagon-generic-builder_vanagon-packaging_generic-builder/2328/

**Comment by skyamgarp** [comment]
Hello,
Here is the vanagon generic pipeline link for agent-runtime-7.x
https://jenkins-platform.delivery.puppetlabs.net/job/platform_vanagon-generic-builder_vanagon-packaging_generic-builder/2333/

File: configs/platforms/el-9-aarch64.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+platform 'el-9-aarch64' do |plat|
+  plat.inherit_from_default
+
+  packages = %w(
+    perl
+    perl-Getopt-Long
```
Comment (joshcooper) [inline configs/platforms/el-9-aarch64.rb:6]: I think the `perl` package pulls in all of the individual `perl-*` modules:

```suggestion
```

**Comment by joshcooper** [comment]
I merged https://github.com/puppetlabs/ci-job-configs/pull/9140 but didn't realize this PR hadn't been merged yet. For future, could you link the two PRs and make it clear that the runtime PR needs to be merged first?

**Review by joshcooper — APPROVED** [review]

**Comment by amitkarsale** [comment]
> I merged [puppetlabs/ci-job-configs#9140](https://github.com/puppetlabs/ci-job-configs/pull/9140) but didn't realize this PR hadn't been merged yet. For future, could you link the two PRs and make it clear that the runtime PR needs to be merged first?

Okay. I will link the two PR's in future.

**Pull request merged**